lib/cog_api/http/base.ex

```diff
@@ -115,7 +114,7 @@ defmodule CogApi.HTTP.Base do
   end
 
   defp make_url(%Endpoint{proto: proto, host: host, port: port,
-                            version: version}, route, params \\ %{}) do
+                            version: version}, route, params_or_id \\ %{}) do
```

Comment:
`params_or_id` is a little weird. What do you think about passing the `id` in as part of the route? so `get(endpoint, "roles/#{id}")` instead of `get(endpoint, "roles", id)`

That way this parameter won't have to serve two functions